File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.kt
class Solution {
    fun maxIceCream(costs: IntArray, coins: Int): Int {
        costs.sort()
        var coin = coins
        var res = 0;
        for(i in costs){
            if( i > coin){
                return res;
            }
            res += 1;
            coin -= i;
        }
        return res;
    }
}

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        std::sort(costs.begin(), costs.end());
        int res = 0;
        for (int i : costs) {
            if (i > coins) {
                return res;
            }
            res += 1;
            coins -= i;
        }
        return res;
    }
};